app/src/main/java/org/simple/clinic/scanid/IndiaNHIDGender.kt

```diff
@@ -0,0 +1,54 @@
+package org.simple.clinic.scanid
+
+import android.os.Parcelable
+import androidx.annotation.VisibleForTesting
+import com.squareup.moshi.FromJson
+import com.squareup.moshi.ToJson
+import kotlinx.parcelize.Parcelize
+import org.simple.clinic.patient.Gender
+import org.simple.clinic.util.room.SafeEnumTypeAdapter
+
+sealed class IndiaNHIDGender : Parcelable {
+
+  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+  object TypeAdapter : SafeEnumTypeAdapter<IndiaNHIDGender>(
+      knownMappings = mapOf(
+          Male to "M",
+          Female to "F",
+      ),
+      unknownStringToEnumConverter = { Unknown(it) },
+      unknownEnumToStringConverter = { (it as Unknown).actualValue }
+  )
+
+  @Parcelize
+  object Male : IndiaNHIDGender()
+
+  @Parcelize
+  object Female : IndiaNHIDGender()
+
+  @Parcelize
+  data class Unknown(val actualValue: String) : IndiaNHIDGender()
+
+  class MoshiTypeAdapter {
+
+    @FromJson
+    fun fromJson(value: String?): IndiaNHIDGender? = TypeAdapter.toEnum(value)
+
+    @ToJson
+    fun toJson(answer: IndiaNHIDGender): String? = TypeAdapter.fromEnum(answer)
+  }
+}
+
+val IndiaNHIDGender.toGender: Gender
+  get() = when (this) {
+    IndiaNHIDGender.Male -> Gender.Male
+    IndiaNHIDGender.Female -> Gender.Female
+    else -> Gender.Unknown(this.toChar)
+  }
+
+val IndiaNHIDGender.toChar: String
+  get() = when (this) {
+    IndiaNHIDGender.Male -> "M"
+    IndiaNHIDGender.Female -> "F"
```

Comment:
Since we technically don't need to handle `Male` and `Female` gender letters, we can just remove this property and use `U` directly in the other property `when` clause, right?

Reply:
Ummm. Fair enough.